benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+            print("WARNING: No orthologous genes found between donor and "
+                  "recipient genome, continuing to novel gene acquisition "
+                  "HGT simulation (if option selected).")
+        if verbose:
+            sys.stdout.write(" done.\n")
+    # simulate novel gene acquisition
+    if float(1-orthologous_rep_prob) > 0.0:
+        if verbose:
+            sys.stdout.write("\tSimulate novel gene acquisition HGTs ...")
+        seq_recip = simulate_novel_acq(genes_donor,
+                                       seq_donor,
+                                       genes_recip,
+                                       seq_recip,
+                                       orthologous_rep_prob,
+                                       percentage_hgts,
+                                       log_f)
+        if verbose:
+            sys.stdout.write(" done.\n")
+    return write_results(
+        genes_donor=genes_donor,
+        donor_genome_fp=donor_genome_fp,
+        genes_recip=genes_recip,
+        recip_genome_fp=recip_genome_fp,
+        seq_donor=seq_donor,
+        seq_recip=seq_recip,
+        output_dir=output_dir)
+
+
+def simulate_genbank(donor_genbank_fp,
+                     recipient_genbank_fp,
+                     output_dir,
+                     percentage_hgts,
+                     orthologous_rep_prob,
+                     log_f,
+                     threads,
+                     verbose=False):
+    """ Simulate HGTs using genuine genomes (GenBank format).
+
+    Parameters
+    ----------
+    donor_genbank_fp: string
+        file path to genome (donor of HGTs) in GenBank format
+    recipient_genbank_fp: string
+        file path to genome (recipient of HGTs) in GenBank format
+    output_dir: string
+        output directory path
+    percentage_hgts: float
+        percentage of HGT genes to simulate (of total genes in recipient
+        genome)
+    orthologous_rep_prob: float
+        rate of orthologous replacement HGTs
+    log_f: file handler
+    threads: integer
+        number of threads to use
+    """
+    if verbose:
+        sys.stdout.write("Parsing donor GenBank record ...\n")
+    seq_donor, genes_donor = extract_genbank(donor_genbank_fp, verbose)
+    if verbose:
+        sys.stdout.write("\tDone.\n")
+    if verbose:
+        sys.stdout.write("Parsing recipient GenBank record ...\n")
+    seq_recip, genes_recip = extract_genbank(recipient_genbank_fp, verbose)
+    if verbose:
+        sys.stdout.write("\tDone.\n")
+
+    return simulate_hgts(
+        seq_donor=seq_donor,
+        genes_donor=genes_donor,
+        seq_recip=seq_recip,
+        genes_recip=genes_recip,
+        donor_genome_fp=donor_genbank_fp,
+        recip_genome_fp=recipient_genbank_fp,
+        output_dir=output_dir,
+        percentage_hgts=percentage_hgts,
+        orthologous_rep_prob=orthologous_rep_prob,
+        log_f=log_f,
+        threads=threads,
+        verbose=verbose)
+
+
+@click.command()
```

Comment:
There's a logical separation here as this is a user interface method, would it make sense to partition this file into "business logic" and "UI logic"?